api/geth_backend.go

```diff
@@ -305,6 +308,90 @@ func (b *GethStatusBackend) startNodeWithAccount(acc multiaccounts.Account, pass
 	return nil
 }
 
+func getKeyFilePathsByAddresses(keyStoreDir string, addresses map[string]struct{}) ([]string, error) {
+	paths := []string{}
+
+	checkFile := func(path string, fileInfo os.FileInfo) error {
+		if fileInfo.IsDir() {
+			return nil
+		}
+
+		rawKeyFile, e := ioutil.ReadFile(path)
+		if e != nil {
+			return fmt.Errorf("invalid account key file: %v", e)
+		}
+
+		var accountKey struct {
+			Address string `json:"address"`
+		}
+		if e := json.Unmarshal(rawKeyFile, &accountKey); e != nil {
+			return fmt.Errorf("failed to read key file: %s", e)
+		}
+
+		address := types.HexToAddress("0x" + accountKey.Address).Hex()
```

Comment:
Not sure I understand here, are you basically normalizing the address by parsing it and converting it back to hex? 
If so I would leave a comment

Reply:
Is this a kind of validation of the address?

Reply:
I actually just copied code from here https://github.com/status-im/status-go/blob/8b61d92d5834215fb06532c527b0bead7b4625d9/account/accounts.go#L142 ` ¯\_(ツ)_/¯`

Reply:
@cammellos @Samyoul let me know if there is a better way to write this code, but at least it works

Reply:
@rasom Would you mind breaking this section of code out into a dedicated `verifyKeyFile()` function and adding some tests specifically for this functionality? 😊 Thank you

Reply:
@Samyoul I added one tiny test for `MigrateKeyStoreDir`, it is a bit puzzling for me how to juggle with slice references (I suppose it would be necessary with `verifyKeyFile` )

Reply:
let me know if you want to see something else there, meanwhile the whole thing is tested manually here https://github.com/status-im/status-react/pull/10847